src/transactions/saga.ts

```diff
@@ -307,3 +313,58 @@ export function* transactionSaga() {
   yield* spawn(watchAddressToE164PhoneNumberUpdate)
   yield* spawn(watchPendingTransactions)
 }
+
+export function* handleTransactionReceiptReceived(
+  txId: string,
+  receipt: TransactionReceipt | CeloTxReceipt,
+  networkId: NetworkId
+) {
+  const tokensById = yield* select((state) => tokensByIdSelector(state, [networkId]))
+
+  // TODO: Receive feeCurrencyId from transaction request.
+  const feeCurrencyId = getTokenId(networkId)
+  const feeTokenInfo = tokensById[feeCurrencyId]
+
+  if (!feeTokenInfo) {
+    Logger.warn(`No information found for token ${feeCurrencyId} in network ${networkId}`)
+  }
+
+  const gasFeeInWei = new BigNumber(receipt.gasUsed.toString()).times(
```

Comment:
nit: it's not in wei, it's in the smallest unit of the feeCurrency.
wei is really [10^-18 ETH](https://ethereum.org/gl/developers/docs/intro-to-ether/#denominations). We should avoid using this naming to avoid bugs. Unless it's really wei we want.

```suggestion
  const gasFeeInSmallestUnit = new BigNumber(receipt.gasUsed.toString()).times(
```